packages/world/ts/deduplicateAbi.ts

```diff
@@ -0,0 +1,24 @@
+import { Abi, AbiItem } from "viem";
+
+function hasNamedInputs(entry: AbiItem) {
+  if (!("inputs" in entry) || !entry.inputs || !entry.inputs.length) {
+    return false;
+  }
+  return entry.inputs.some((input) => input.name);
+}
+
+// during deduplication, favor ABI items with named inputs
+export function deduplicateAbi(abi: Abi) {
+  const uniqueEntries = new Map();
+
+  for (const entry of abi) {
+    const key = "name" in entry ? `${entry.type}_${entry.name}` : entry.type;
```

Comment:
I think this is assuming each function is unique by name but Solidity supports overloads 

might make use of abitype or viem helpers to convert this to a function signature before comparing

Reply:
Ah that's very good point, and a massive oversight. Adjusted to take into this into account.